### ISSUE TITLE
Add a list of available viable providers per chain

### ISSUE THREAD

**Issue by bbenligiray** [api3dao/chains#103]
We need to maintain a list of viable providers for each chain (where viable means 99%+ availability and good support of all endpoints). This is a proposal to do that in this repo. Otherwise, this will be done in another package that imports @api3/chains.

**Comment by andreogle** [comment]
This repo sounds like the correct place to me. Do you have a data structure in mind?

We could change `providerUrl` to an array of strings but that also raises some questions about which one someone should pick. Should they pick the first/last/random one? Or should this be in addition to the existing `providerUrl` field?

```ts
{
  providerUrls: [
    'https://rpc-1.xyz',
    'https://rpc-2.xyz'
  ]
}
```

**Comment by bbenligiray** [comment]
This needs to cover paid providers as well, in which case it won't be possible to specify a provider URL. (Arguably we can do something like `https://mainnet.infura.io/v3/{YOUR-API-KEY}` like in [ethers](https://docs.ethers.org/v5/api/providers/#providers-getDefaultProvider) but that's just busywork imo.) I was thinking of something like
```
{
  "providers": [
    {
      "alias": "alchemy",
      "homepageUrl": "https://www.alchemy.com/"
    },
    {
      "alias": "public",
      "rpcUrl": "https://rpc.gnosischain.com"
    }
  ]
}
```
This would replace `providerUrl`.

**Comment by bbenligiray** [comment]
We can expose the `monitor.reblok.io` status through an API to the CI of this repo such as
```
{
  "100": [
    {
      "alias": "alchemy",
      "healthy": true
    },
    {
      "alias": "public",
      "healthy": true
    }
  ]
}
```
where health is a function of many factors (is the response rate over 95%, does it respond to `eth_getLogs`, what is the repsonse time for a 1,000,000 gas static call, etc.), which would ultimately indicate if it does more harm than good if it was used as a redundancy.
This would then supersede [the daily provider ping](https://github.com/api3dao/chains/issues/70).

@bdrhn9 for visibility

**Comment by andreogle** [comment]
Closing this again as it's included in the recently released v4.0.0
